packages/gatsby/src/utils/webpack-error-utils.ts

```diff
@@ -11,42 +12,91 @@ type Stage = keyof typeof stageCodeToReadableLabel
 type StageLabel = typeof stageCodeToReadableLabel[Stage]
 
 interface ITransformedWebpackError {
-  id: "98123"
+  id: string
   filePath?: string
   location?: {
     start: string
   }
   context: {
     stage: Stage
     stageLabel: StageLabel
-    message?: string
+    sourceMessage?: string
+    [key: string]: string | boolean | undefined
   }
 }
 const transformWebpackError = (
   stage: keyof typeof stageCodeToReadableLabel,
   webpackError: any
 ): ITransformedWebpackError => {
-  return {
-    id: `98123`,
-    filePath: webpackError?.module?.resource,
-    location:
-      webpackError?.module?.resource && webpackError?.error?.loc
-        ? {
-            start: webpackError.error.loc,
-          }
-        : undefined,
-    context: {
-      stage,
-      stageLabel: stageCodeToReadableLabel[stage],
-      message: webpackError?.error?.message || webpackError?.message,
+  const handlers = [
+    {
+      regex: /Can't resolve '(.*?)' in '(.*?)'/m,
+      cb: (match): IMatch => {
+        return {
+          id: `98124`,
+          context: {
+            sourceMessage: match[0],
+            packageName: match[1],
+          },
+        }
+      },
     },
+  ]
 
-    // We use original error to display stack trace for the most part.
-    // In case of webpack error stack will include internals of webpack
-    // or one of loaders (for example babel-loader) and doesn't provide
-    // much value to user, so it's purposely omitted.
-    // error: webpackError?.error || webpackError,
+  // We use original error to display stack trace for the most part.
+  // In case of webpack error stack will include internals of webpack
+  // or one of loaders (for example babel-loader) and doesn't provide
+  // much value to user, so it's purposely omitted.
+  // error: webpackError?.error || webpackError,
+  const webpackMessage = webpackError?.error?.message || webpackError?.message
+
+  let structured: ITransformedWebpackError | undefined
+
+  for (const { regex, cb } of handlers) {
+    const matched = webpackMessage?.match(regex)
+    if (matched) {
+      const match = cb(matched)
+      structured = {
+        id: match.id,
+        filePath: webpackError?.module?.resource,
+        location:
+          webpackError?.module?.resource && webpackError?.error?.loc
+            ? {
+                start: webpackError.error.loc,
+              }
+            : undefined,
+        context: {
+          sourceMessage: match.context.sourceMessage,
+          packageName: match.context.packageName,
+          stage,
+          stageLabel: stageCodeToReadableLabel[stage],
+        },
```

Comment:
Can we do common things for all errors in one place (`filePath`, `location`, `context.sourceMessage`, `context.stage`, `context.stageLabel`) and spread it? Just to avoid copy&paste because it's too easy to fix/change something in one place and don't handle the other?